client_end_user/shared/src/commonMain/kotlin/domain/gateway/IRestaurantRemoteGateway.kt

```diff
@@ -1,8 +1,12 @@
 package domain.gateway
 
 import domain.entity.Cuisine
+import domain.entity.Order
+import domain.entity.Taxi
 
 interface IRestaurantRemoteGateway {
     suspend fun getCuisines(): List<Cuisine>
-
+    suspend fun getTaxiOnTheWay(): Taxi?
```

Comment:
Why nulable ? , I think you should handle nulability on gateway and throw an exception there